### DISCUSSION TITLE
How to install large programs?

### DISCUSSION THREAD

**Discussion by richelbilderbeek** [community/community#27213] [Actions]
<p>I depend on <a href="https://www.ncbi.nlm.nih.gov/tools/cobalt/re_cobalt.cgi" rel="noopener nofollow ugc">a program (COBALT)</a> that needs some large files to run, such as the +250MB file at<br>
<code>ftp://ftp.ncbi.nlm.nih.gov/pub/agarwala/cobalt/cdd.rps</code>. In total, one needs approx 1GB to use the program.</p>
<p>I understand GHA cancels the download of such a big install, such as <a href="https://github.com/richelbilderbeek/cobaltr/runs/1517194770?check_suite_focus=true#step:11:649" rel="noopener nofollow ugc">here</a>:</p>
<pre><code class="lang-auto">trying URL 'ftp://ftp.ncbi.nlm.nih.gov/pub/agarwala/cobalt/cdd.rps'
Content type 'unknown' length 296721936 bytes (283.0 MB)
Error: The operation was canceled.
</code></pre>
<p>Still, I do need that program. How can I convince GHA to download the files I need?</p>
<p>As these are file needed for installation only, I prefer them not to be artifacts, so not to bother GitHub with storing large useless files.</p>
<p>Thanks and cheers, Richel Bilderbeek</p>

**Comment by bboles** [comment]
<p>Self-hosted runner may be the best option.  That way you can setup any tooling you need outside of Github Actions.  Otherwise you will spend more time trying to navigate the restrictions that the provided runners impose.</p>

**Comment by Simran-B** [comment]
<p>How long does the download take?</p>
<p>I have no issues with using a 1 GB Docker image as <code>container</code>, maybe that’s an option for you too. Pulling from Docker Hub is really fast. You can also try GHCR.</p>

**Comment by richelbilderbeek** [comment, marked as answer]
<p>I noticed that the problem lies not with GitHub Actions, but with the FTP server where I download the files from.</p>
<p>Now I conclude that GitHub Actions does large downloads fine, if FTP servers let it.</p>
<p>Solved :+1:</p>
